### PR TITLE
LEAN-3304

### PR DESCRIPTION
**Pull request by mnatsheh**

### CODE REVIEW

File: src/postgres-data/PsSource.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default class PsSource implements StoreDataSourceStrategy {
     }
     if (state.manuscriptID && state.projectID) {
       this.data = await buildData(state.projectID, state.manuscriptID, this.api)
-      this.utilities = buildUtilities(() => this.data, this.api, setState)
+      this.utilities = buildUtilities(() => this.data, setState, this.api)
```
Comment (mbartenev-atypon) [inline src/postgres-data/PsSource.ts:45]: I wonder what is the purpose of this change?

File: src/postgres-data/buildUtilities.ts [hunk 2]
```diff
@@ -10,81 +10,46 @@
...
-          return false
-        })
-    } else {
-      return Promise.reject(false)
+    if (state.projectID && state.manuscriptID) {
+      try {
+        const filtered = models.filter(
+          (m) => m.objectType !== ObjectTypes.Project
+        )
+        await api.saveProject(state.projectID, filtered)
```
Comment (mbartenev-atypon) [inline src/postgres-data/buildUtilities.ts:46]: do we not need to make sure to send all the models for that document, won't API delete the models that we didn't send?

File: src/components/projects/CommentList.tsx [hunk 3]
```diff
@@ -51,11 +51,10 @@ export const CommentList: React.FC<Props> = ({ editor }) => {
       saveTrackModel,
     },
   ] = useStore((store) => ({
-    notes: store.notes,
     user: store.user,
     collaborators: store.collaborators || new Map<string, UserProfile>(),
     collaboratorsById: store.collaboratorsById,
-    keywords: store.keywords,
+    keywords: new Map(),
```
[the remark this answers is not in the document]
Comment (mnatsheh) [reply src/components/projects/CommentList.tsx:57]: I wanted to limit changes to components as much as possible. This component needs cleanup anyway, so we can get rid of this completely as part of that.

File: src/postgres-data/buildUtilities.ts (hunk 2 above)
Comment (mnatsheh) [reply src/postgres-data/buildUtilities.ts:46]: This should have the same logic as before, just using async/await, did I miss something with the conversion?

File: src/postgres-data/buildUtilities.ts [hunk 4]
```diff
@@ -210,94 +122,31 @@ const buildUtilities = (
...
+    }
+    const prevManuscript = state.modelMap.get(state.manuscriptID)
+    await saveModel({
+      ...prevManuscript,
+      ...manuscriptData,
+    })
   }
 
-  // async ( model: T | Build<T> | Partial<T>
   const bulkUpdate = async (
```
[the remark this answers is not in the document]
Comment (mnatsheh) [reply src/postgres-data/buildUtilities.ts]: I updated the code to eliminate this and have everything depend on `saveModels`. I had to clean up `use-handle-snapshot` by moving `saveSnapshot` and `saveDoc` to `buildUtilities`.

File: src/postgres-data/buildUtilities.ts [hunk 5]
```diff
@@ -210,94 +122,31 @@ const buildUtilities = (
   }
 
   const saveManuscript = async (manuscriptData: Partial<Manuscript>) => {
```
[the remark this answers is not in the document]
Comment (mbartenev-atypon) [reply src/postgres-data/buildUtilities.ts]: Ok. We can have a look at it later on

File: src/components/projects/CommentList.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbartenev-atypon) [reply src/components/projects/CommentList.tsx:57]: Are you planning to do that cleanup yourself? If not, please add comment in it that this has to be cleaned up in the store too

File: src/postgres-data/buildUtilities.ts (hunk 2 above)
Comment (mbartenev-atypon) [reply src/postgres-data/buildUtilities.ts]: No, you are right. That's how it used to be

File: src/postgres-data/buildUtilities.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbartenev-atypon) [reply src/postgres-data/buildUtilities.ts]: Thanks

**Review by mbartenev-atypon — APPROVED** [review]

File: src/components/projects/CommentList.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mnatsheh) [reply src/components/projects/CommentList.tsx:58]: Added a comment.

File: src/postgres-data/PsSource.ts (hunk 1 above)
Comment (mnatsheh) [reply src/postgres-data/PsSource.ts:45]: Just wanted to make the `api` param at the end to make it consistent with the other call. Not necessary though.

File: src/postgres-data/buildUtilities.ts [hunk 6]
```diff
@@ -11,127 +11,99 @@
...
     }
 
-    const modelMap = new Map(data.modelMap)
-    modelMap.set(containedModel._id, newModel)
+    const modelMap = new Map<string, Model>()
 
-    // data.modelMap.set(containedModel._id, newModel)
+    for (const [id, model] of state.modelMap) {
+      const type = model.objectType as ObjectTypes
+      if (!excludeTypes.has(type) && !excludeIDs?.has(id)) {
```
Comment (asouqi) [inline src/postgres-data/buildUtilities.ts:91]: @mnatsheh with this check we are not going to save manuscript model! and if you recall `/project/../manuscript/..` after save will not get `MPManuscript`. so should we use a different endpoint to get manuscript?

Comment (mnatsheh) [reply src/postgres-data/buildUtilities.ts:91]: @asouqi , we should update this to save the manuscript. Feel free to open a PR or file a ticket and I'll take a look.
